Cloudflare types for `env` no longer work

### Is there an existing issue for this?

- [x] I have checked for existing issues https://github.com/getsentry/sentry-javascript/issues
- [x] I have reviewed the documentation https://docs.sentry.io/
- [x] I am using the latest SDK release https://github.com/getsentry/sentry-javascript/releases

### How do you use Sentry?

Sentry Saas (sentry.io)

### Which SDK are you using?

@sentry/cloudflare

### SDK Version

10.26.0

### Framework Version

_No response_

### Link to Sentry event

_No response_

### Reproduction Example/SDK Setup

Repository: https://github.com/Jiralite/sentry-cloudflare-env-reproduction

There is a very suspicious pull request, #18064, that seems responsible for this as reverting that definition in 10.26.0 correctly infers `env` (this means versions 10.23.0 onwards breaks the types).

### Steps to Reproduce

1. Run TypeScript

### Expected Result

`env` is inferred.

### Actual Result

```
source/index.ts:7:44 - error TS18046: 'env' is of type 'unknown'.

7 export default withSentry((env) => ({ dsn: env.SENTRY_DATA_SOURCE_NAME, sendDefaultPii: true }), {
                                             ~~~

source/index.ts:8:2 - error TS2322: Type '(_: Request<unknown, IncomingRequestCfProperties<unknown>>, env: Env) => Response' is not assignable to type 'ExportedHandlerFetchHandler<unknown, unknown>'.
  Types of parameters 'env' and 'env' are incompatible.
    Type 'unknown' is not assignable to type 'Env'.

8  fetch(_, env) {
   ~~~~~

  node_modules/.pnpm/@cloudflare+workers-types@4.20251121.0/node_modules/@cloudflare/workers-types/2023-07-01/index.d.ts:520:3
    520   fetch?: ExportedHandlerFetchHandler<Env, CfHostMetadata>;
          ~~~~~
    The expected type comes from property 'fetch' which is declared here on type 'ExportedHandler<unknown, unknown, unknown>'


Found 2 errors in the same file, starting at: source/index.ts:7
```

### Additional Context

_No response_

### Priority

<sub>React with 👍 to help prioritize this issue. Please use comments to provide useful context, avoiding `+1` or `me too`, to help us triage it.</sub>

--- a/packages/cloudflare/src/withSentry.ts
+++ b/packages/cloudflare/src/withSentry.ts
@@ -1,12 +1,13 @@
+import type { env } from 'cloudflare:workers';
 import { setAsyncLocalStorageAsyncContextStrategy } from './async';
 import type { CloudflareOptions } from './client';
 import { isInstrumented, markAsInstrumented } from './instrument';
-import { getHonoIntegration } from './integrations/hono';
 import { instrumentExportedHandlerEmail } from './instrumentations/worker/instrumentEmail';
 import { instrumentExportedHandlerFetch } from './instrumentations/worker/instrumentFetch';
 import { instrumentExportedHandlerQueue } from './instrumentations/worker/instrumentQueue';
 import { instrumentExportedHandlerScheduled } from './instrumentations/worker/instrumentScheduled';
 import { instrumentExportedHandlerTail } from './instrumentations/worker/instrumentTail';
+import { getHonoIntegration } from './integrations/hono';
 
 /**
  * Wrapper for Cloudflare handlers.
@@ -20,7 +21,7 @@ import { instrumentExportedHandlerTail } from './instrumentations/worker/instrum
  * @returns The wrapped handler.
  */
 export function withSentry<
-  Env = unknown,
+  Env = typeof env,
   QueueHandlerMessage = unknown,
   CfHostMetadata = unknown,
   T extends ExportedHandler<Env, QueueHandlerMessage, CfHostMetadata> = ExportedHandler<

--- a/packages/cloudflare/test/withSentry.test.ts
+++ b/packages/cloudflare/test/withSentry.test.ts
@@ -7,18 +7,21 @@ import { withSentry } from '../src/withSentry';
 import { markAsInstrumented } from '../src/instrument';
 import * as HonoIntegration from '../src/integrations/hono';
 
-type HonoLikeApp<Env = unknown, QueueHandlerMessage = unknown, CfHostMetadata = unknown> = ExportedHandler<
+declare global {
+  namespace Cloudflare {
+    interface Env {
+      SENTRY_DSN: string;
+    }
+  }
+}
+
+type HonoLikeApp<Env = Cloudflare.Env, QueueHandlerMessage = unknown, CfHostMetadata = unknown> = ExportedHandler<
   Env,
   QueueHandlerMessage,
   CfHostMetadata
 > & {
   onError?: () => void;
   errorHandler?: (err: Error) => Response;
-};
-
-const MOCK_ENV = {
-  SENTRY_DSN: 'https://public@dsn.ingest.sentry.io/1337',
-  SENTRY_RELEASE: '1.1.1',
 };
 
 describe('withSentry', () => {
@@ -33,7 +36,7 @@ describe('withSentry', () => {
       const handleHonoException = vi.fn();
       vi.spyOn(HonoIntegration, 'getHonoIntegration').mockReturnValue({ handleHonoException } as any);
 
-      const honoApp = {
+      const honoApp: HonoLikeApp = {
         fetch(_request, _env, _context) {
           return new Response('test');
         },
@@ -41,7 +44,7 @@ describe('withSentry', () => {
         errorHandler(err: Error) {
           return new Response(`Error: ${err.message}`, { status: 500 });
         },
-      } satisfies HonoLikeApp<typeof MOCK_ENV>;
+      };
 
       withSentry(env => ({ dsn: env.SENTRY_DSN }), honoApp);
 
@@ -59,13 +62,13 @@ describe('withSentry', () => {
 
       const error = new Error('test hono error');
 
-      const honoApp = {
+      const honoApp: HonoLikeApp = {
         fetch(_request, _env, _context) {
           return new Response('test');
         },
         onError() {},
         errorHandler: originalErrorHandlerSpy,
-      } satisfies HonoLikeApp<typeof MOCK_ENV>;
+      };
 
       withSentry(env => ({ dsn: env.SENTRY_DSN }), honoApp);
 
@@ -86,13 +89,13 @@ describe('withSentry', () => {
 
       markAsInstrumented(originalErrorHandler);
 
-      const honoApp = {
+      const honoApp: HonoLikeApp = {
         fetch(_request, _env, _context) {
           return new Response('test');
         },
         onError() {},
         errorHandler: originalErrorHandler,
-      } satisfies HonoLikeApp<typeof MOCK_ENV>;
+      };
 
       withSentry(env => ({ dsn: env.SENTRY_DSN }), honoApp);
 

